server/history/schema/history.schema.ts

```diff
@@ -0,0 +1,44 @@
+import { Prop, Schema, SchemaFactory } from "@nestjs/mongoose";
+import * as mongoose from "mongoose";
+import { User } from "../../users/schemas/user.schema";
+import { HistoryType } from "../../claim/claim.service";
+
+export type HistoryDocument = History & mongoose.Document;
+
+@Schema({ toObject: {virtuals: true}, toJSON: {virtuals: true} })
+export class History {
+    @Prop({
+        type: mongoose.Types.ObjectId,
+        required: true,
+        refPath: "onModel",
+    })
+    targetId: mongoose.Types.ObjectId;
+
+    @Prop({
+        required: true,
+        enum: ["Claim", "Personality"],
+    })
+    targetModel: string;
+
+    @Prop({
+      type: mongoose.Types.ObjectId,
+      required: true,
+      ref: "User",
+    })
+    user: User;
+        
+    @Prop({
+      required: true,
+    })
+    type: HistoryType
+
+    @Prop({
+      type: Object,
+      required: true,
+    })
+    details: object
+}
+
+export const HistorySchema = SchemaFactory.createForClass(History);
+
+//ter uma validação dos tipos, caso seja um tipo create, é obrigatório um after
```

Comment:
That's a TODO for another patch? Could you write it in English, please?

Reply:
I think i'll resolve in this patch, just to reminder myself